src/main/java-templates/org/postgresql/ds/PGConnectionPoolDataSource.java

```diff
@@ -29,7 +29,7 @@
  * @author Aaron Mulder (ammulder@chariotsolutions.com)
  */
 public class PGConnectionPoolDataSource
-    extends @CONN_POOL_DS_CLASS@
+    extends ${CONN_POOL_DS_CLASS}
```

Comment:
@mkarg It's easier to explain with a code sample :-) 

Here is an example of the pre-processing that currently occurs in the build. In this case the class it extends can be either src/main/java/org/postgresql/ds/jdbc23/AbstractJdbc23PoolingDataSource.java or src/main/java/org/postgresql/ds/jdbc4/AbstractJdbc4PoolingDataSource.java. If the build is for JDBC4 or above the latter is chosen.

If we support JDBC4 as a minimum I believe many of these pre-processing steps can be removed. This would leave the Driver class which I'll annotate separately.